platform/kube/inject/initializer.go

```diff
@@ -88,242 +62,202 @@ type InitializerOptions struct {
 // istioSidecarAnnotationPolicyKey).
 type Initializer struct {
 	clientset   kubernetes.Interface
-	mesh        *proxyconfig.ProxyMeshConfig
 	controllers []cache.Controller
-	options     InitializerOptions
-	params      Params
+	config      *Config
+}
+
+var (
+	kinds = []struct {
+		groupVersion schema.GroupVersion
+		obj          runtime.Object
+		resource     string
+		apiPath      string
+	}{
+		{v1.SchemeGroupVersion, &v1.ReplicationController{}, "replicationcontrollers", "/api"},
+
+		{v1beta1.SchemeGroupVersion, &v1beta1.Deployment{}, "deployments", "/apis"},
+		{v1beta1.SchemeGroupVersion, &v1beta1.DaemonSet{}, "daemonsets", "/apis"},
+		{v1beta1.SchemeGroupVersion, &v1beta1.ReplicaSet{}, "replicasets", "/apis"},
```

Comment:
You added a new type `replicasets` here, but the file https://github.com/istio/pilot/blob/master/test/integration/testdata/initializer-config.yaml.tmpl was not updated

Reply:
The list here is only what the sidecar-initializer supports. The resource types in initializer-config.yaml.tmpl are what is actually configured. I removed the non-deployment types in the e2e test to make this more clear.